### ISSUE TITLE
Mantine Switch should follow accessibility best practices

### ISSUE THREAD

**Issue by ocshawn** [mantinedev/mantine#5699]
Current Switch component fails accessibility checks for using an empty label to land user clicks on a checkbox. If user also uses the label prop a second label is created with the same `for` target which is also not allowed. 

Consider reworking the component to behave more like the [W3C APG example ](https://www.w3.org/WAI/ARIA/apg/patterns/switch/examples/switch/)

**Comment by rilrom** [comment]
@ocshawn good points, I'll attempt to put together a suitable solution for this.

**Comment by ocshawn** [comment]
@rilrom im also working on a fix (but im new to the community) so we will see how it goes

**Comment by ocshawn** [comment]
@rilrom  is it ok if i create a PR for v6 using your fix as a basis as we have not upgraded from v6 yet?

**Comment by ocshawn** [comment]
@rtivital Could we get a v6 release that incorporates this fix, we are working on upgrading to 7 but currently still running 6
